crates/snp_validation/src/lib.rs

```diff
@@ -39,9 +45,13 @@ flags! {
         SMT = 1 << 0,
     }
 
+    /// These items are mutually exclusive
+    #[derive(Deserialize, Serialize)]
     pub enum PlatformInfoFlags: u8 {
+        /// Secure Memory Encryption
+        SME = 1 << 0,
+        /// Transparent Secure Memory Encryption
         TSME = 1 << 1,
-        SMT = 1 << 0,
```

Comment:
Where did you get that info?

I have from `SEV Secure Nested Paging Firmware ABI Specification Revision 1.51`:

Table 22. Structure of the PLATFORM_INFO Field

|Byte Offset | Bits   | Name | Description |
|---------------|--------|---------|----------------|
| 0h              | 63:2 | -                | Reserved. |
|                   |      1 | TSME_EN | Indicates that TSME is enabled in the system |
|                   |      0 |   SMT_EN | Indicates that SMT is enabled in the system. |
```